SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "Utils/LayerInfo.h"
+#include "Utils/SpatialActorUtils.h"
+
+#include "Templates/Tuple.h"
+
+DEFINE_LOG_CATEGORY(LogLayeredLBStrategy);
+
+ULayeredLBStrategy::ULayeredLBStrategy()
+	: Super()
+{
+}
+
+ULayeredLBStrategy::~ULayeredLBStrategy()
+{
+	for (const auto& Elem : LayerNameToLBStrategy)
+	{
+		Elem.Value->RemoveFromRoot();
+	}
+}
+
+void ULayeredLBStrategy::Init()
+{
+	Super::Init();
+
+	VirtualWorkerId CurrentVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID + 1;
+
+	if (const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>())
```

Comment:
Perhaps we could return early if Settings->bEnableUnrealLoadBalancer is false so that we can remove the large amount of indentation?

Reply:
Actually, it can just be a check.

Reply:
Wonder if we should check settings too? TBF I think there are lots of places in the code we just use the pointer directly.

Reply:
I think things would have blown up way before this if Settings wasn't valid.